File: Data Structures & Algorithms/permutations-ii/submission-1.py
class Solution:
    def permuteUnique(self, nums: List[int]) -> List[List[int]]:
        used = set()
        perm = []
        perms = []

        def dfs(i):
            if len(perm) == len(nums) and perm not in perms:
                perms.append(perm.copy())
                return
            
            for j in range(len(nums)):
                if j in used:
                    continue
                
                perm.append(nums[j])
                used.add(j)
                dfs(j + 1)
                perm.pop()
                used.remove(j)

        dfs(0)
        return perms

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-1.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int p = 0;
        int d = 1;

        while (p < nums.size()) {
            while (d < nums.size() && nums[d] == nums[p]) {
                nums.erase(nums.begin() + d);
            };

            p = d;
            d++;
        };

        return nums.size();
    };
};